job_templates/ci_job.xml.em

```diff
@@ -167,6 +175,9 @@ echo "Using args: $CI_ARGS"
 echo "# END SECTION"
 
 @[if os_name in ['linux', 'linux-aarch64']]@
+sed -i "s+^FROM.*$+FROM ubuntu:$CI_UBUNTU_DISTRO+" linux_docker_resources/Dockerfile
```

Comment:
Yep, if we need to customise the ARM base image again, I imagine we'd add another sed statement that is more specific.

Just adding on a couple of points that were missing on the original thread (mostly just "info dump"):
- I don't think we've been using custom OSRF arm images since https://github.com/ros2/ci/pull/47/files#diff-526f24c7be4a93f0e5699f48a9f1656fR180.
- We prefer to use official ubuntu docker images, and don't need arm-specific dockerfile workarounds anymore because ubuntu images provide architecture-specific support now. That change in our CI happened with the swap to bionic, but it holds true on xenial too.

Reply:
Some clarifications:

We still don't use official docker images, that is why we still need to sed the Dockerfiles. My goal is to actually stop relying on third party images to use the official ones: https://github.com/ros2/ci/pull/148/commits/1ac0506d878d7dead1bdd033ec6c874afbdc88e1

When I brought this up, we had a discussion about (in the future) providing a way for people to run our CI without having access to arm servers (or even us if we want to start building armhf again), at which point it should be easy to use the osrf custom images instead.
That's why I'm a bit hesitant about introducing another set of `sed` commands that would overwrite the entire `FROM` statement

Reply:
Yeah I meant "we use the official ones now" referring to "as of the bionic change" :smiley:

OK, that's where the custom arm images would come back into it. I hadn't made that connection from just looking at our setup but it makes sense for others. Thanks for clarifying :)